### PR TITLE
Add explanation of STATIC_ASSERTS in pokemon.c

### PR DESCRIPTION
**Pull request by pkmnsnfrn**
Original discussion: https://discord.com/channels/419213663107416084/774393519569502268/1220032007225217046

I don't feel strongly about how this is worded, but I think an explanation is a good idea. If this turns out to be a common / bigger issue, we can circle back.

### CODE REVIEW

**Comment by mrgriffin** [comment]
I think it would be worth adding a brief note that unless there's `unused` data after your member, naively growing the member will likely break the save... and perhaps that a third alternative would be to repurpose some IDs that have never been used by your game?

I could imagine established projects choosing that third path. (There's also a fourth path of using a non-contiguous bit, and then combining them in `GetMonData`/separating them in `SetMonData`, but that's probably too wordy and fiddly to explain)

**Comment by pkmnsnfrn** [comment]
@mrgriffin could you clarify IDs that you think developers _wouldn't_ use? I just want to make sure I understand it before giving an example (and yeah, I'm skipping path 4 lmao, if you're skilled enough to do that, you're not reading my comment at all)

**Comment by mrgriffin** [comment]
Probably the most common examples will be key items—e.g. Gold Teeth or Magma Emblem. But also potentially things like a developer going "well I never want Tera Shards", or "I don't have Giratina" and so not having Griseous Orb and/or Core.

Similarly, if they've decided to permanently turn off species then they might decide to repurpose those IDs.

**Review by mrgriffin — APPROVED** [review]
tysm!
